web/src/localstorage.js

```diff
@@ -68,12 +72,21 @@ const ls = {
     // the pattern given by `name`.
     removeDataRegexWithCondition(version, regex, condition_checker) {
         const key_regex = new RegExp(this.formGetter(version, regex));
-        const keys = Object.keys(localStorage).filter(
-            (key) => key_regex.test(key) && condition_checker(localStorage.getItem(key)),
-        );
+        let keys = [];
+        try {
+            keys = Object.keys(localStorage).filter(
+                (key) => key_regex.test(key) && condition_checker(localStorage.getItem(key)),
+            );
+        } catch {
+            // Do nothing if we fail to fetch the local storage
+        }
```

Comment:
Similarly, we probably don't want to eat exceptions from `condition_checker` -- so we may want to just wrap a `raw_keys = ...` bit in the try/catch, and have the filter/test/condition stuff outside it (being a noop on an empty list).

Reply:
But we do want to wrap the `localstorage.getItem`, which can also fail separately from the `Object.keys(localstorage)`